zerver/middleware.py

```diff
@@ -657,12 +657,8 @@ def alter_content(request: HttpRequest, content: bytes) -> bytes:
 
 
 class FinalizeOpenGraphDescription(MiddlewareMixin):
-    def process_response(
-        self, request: HttpRequest, response: StreamingHttpResponse
-    ) -> StreamingHttpResponse:
-
+    def process_response(self, request: HttpRequest, response: HttpResponse) -> HttpResponse:
```

Comment:
Middleware can’t know whether the response is `HttpResponse`; it could be `FileResponse`. So middleware should accept `HttpResponseBase`, and runtime-check any downcasts as appropriate.

(This would be caught by mypy if django-stubs knew about `MiddlewareMixin.process_response`—that should be added upstream.)